src/Visitor/Php/Symfony/Constraint.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Translation\Extractor\Visitor\Php\Symfony;
+
+use PhpParser\Node;
+use PhpParser\NodeVisitor;
+use Translation\Extractor\Visitor\Php\BasePHPVisitor;
+
+final class Constraint extends BasePHPVisitor implements NodeVisitor
+{
+    const CONSTRAINT_CLASS_NAMES = [
+        'AbstractComparison',
+        'All',
+        'Bic',
+        'Blank',
+        'Callback',
+        'CardScheme',
+        'Choice',
+        'Collection',
+        'Composite',
+        'Count',
+        'Country',
+        'Currency',
+        'Date',
+        'DateTime',
+        'DisableAutoMapping',
+        'DivisibleBy',
+        'Email',
+        'EnableAutoMapping',
+        'EqualTo',
+        'Existence',
+        'Expression',
+        'File',
+        'GreaterThan',
+        'GreaterThanOrEqual',
+        'GroupSequence',
+        'GroupSequenceProvider',
+        'Iban',
+        'IdenticalTo',
+        'Image',
+        'Ip',
+        'Isbn',
+        'IsFalse',
+        'IsNull',
+        'Issn',
+        'IsTrue',
+        'Json',
+        'Language',
+        'Length',
+        'LessThan',
+        'LessThanOrEqual',
+        'Locale',
+        'Luhn',
+        'Negative',
+        'NegativeOrZero',
+        'NotBlank',
+        'NotCompromisedPassword',
+        'NotEqualTo',
+        'NotIdenticalTo',
+        'NotNull',
+        'NumberConstraintTrait',
+        'Optional',
+        'Positive',
+        'PositiveOrZero',
+        'Range',
+        'Regex',
+        'Required',
+        'Time',
+        'Timezone',
+        'Traverse',
+        'Type',
+        'Unique',
+        'Url',
+        'Uuid',
+        'Valid',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function beforeTraverse(array $nodes): ?Node
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enterNode(Node $node): ?Node
+    {
+        if (!$node instanceof Node\Expr\New_) {
+            return null;
+        }
+
+        $className = $node->class;
+        if (!$className instanceof Node\Name) {
+            return null;
+        }
+
+        $parts = $className->parts;
+        $isConstraintClass = false;
+        foreach ($parts as $part) {
+            if (\in_array($part, self::CONSTRAINT_CLASS_NAMES)) {
+                $isConstraintClass = true;
```

Comment:
If we found the constraint class - should we continue iterating further or stop iteration? I just don't see the value in iterating further

Reply:
Unfortunately if I write `Assert\NotBlank` instead of just NotBlank the "parts" would contain Assert and NotBlank so I need to iterate all of them.

Reply:
Can't part of finding an appropiate constraint class get caught by checking against extending `Symfony\Component\Validator\Constraint`?

Reply:
How can I check against extending a Constraint class, that would be useful!

Reply:
A `is_subclass_of` function call can get used for that https://3v4l.org/jcgXX

Reply:
Yes, I know that but I can't use `is_subclass_of` because in the $part I don't have the fully qualified class name but just the "string" of the class name (e.g. "NotBlank").